src/integration/integration_test.go

```diff
@@ -0,0 +1,123 @@
+package integration_test
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"os"
+	"os/exec"
+	"text/template"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/onsi/gomega/gbytes"
+	"github.com/onsi/gomega/gexec"
+	colmetricspb "go.opentelemetry.io/proto/otlp/collector/metrics/v1"
+	metricspb "go.opentelemetry.io/proto/otlp/metrics/v1"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	_ "google.golang.org/grpc/encoding/gzip"
+)
+
+var _ = Describe("OTel Collector", func() {
+	var otelCollectorSession *gexec.Session
+	var ctx context.Context
+	var testOtelReceiver *grpc.Server
+	var fakeMetricsServiceServer FakeMetricsServiceServer
+	var msc colmetricspb.MetricsServiceClient
+	var otelConfigVars OTELConfigVars
+	var otelConfigPath string
+
+	BeforeEach(func() {
+		otelConfigVars = NewOTELConfigVars()
+
+		lis, err := net.Listen("tcp", fmt.Sprintf(":%d", otelConfigVars.EgressOTLPPort))
+		Expect(err).NotTo(HaveOccurred())
+
+		tlsCert := otelConfigVars.TLSCert()
+		testOtelReceiver = grpc.NewServer(grpc.Creds(credentials.NewServerTLSFromCert(&tlsCert)))
+
+		fakeMetricsServiceServer = NewFakeMetricsServiceServer()
+		colmetricspb.RegisterMetricsServiceServer(testOtelReceiver, fakeMetricsServiceServer)
+		go testOtelReceiver.Serve(lis)
+
+		conn, err := grpc.NewClient(fmt.Sprintf("127.0.0.1:%d", otelConfigVars.IngressOTLPPort),
+			grpc.WithTransportCredentials(otelConfigVars.CaAsGrpcTransportCredentials()),
+		)
+		Expect(err).NotTo(HaveOccurred())
+		msc = colmetricspb.NewMetricsServiceClient(conn)
+	})
+
+	JustBeforeEach(func() {
+		var err error
+		configyml, err := os.ReadFile(fmt.Sprintf("testdata/%s", otelConfigPath))
+		Expect(err).NotTo(HaveOccurred())
+		t, err := template.New("test").Parse(string(configyml))
+		Expect(err).NotTo(HaveOccurred())
+
+		buf := new(bytes.Buffer)
+		err = t.Execute(buf, otelConfigVars)
+		Expect(err).NotTo(HaveOccurred())
+		fmt.Println(buf.String())
+		os.WriteFile(TempFilePath("config.yml"), buf.Bytes(), 0660)
+
+		ctx = context.TODO()
+		cmd := exec.CommandContext(ctx, componentPaths.Collector, fmt.Sprintf("--config=file:%s", TempFilePath("config.yml")))
+		otelCollectorSession, err = gexec.Start(cmd, GinkgoWriter, GinkgoWriter)
+		Expect(err).NotTo(HaveOccurred())
+		Eventually(ctx, otelCollectorSession.Err).Should(gbytes.Say(`Everything is ready. Begin running and processing data.`))
```

Comment:
Not sure that we want this Eventually to share a time horizon with the tests. Shouldn't this be acting as a blocker to ensure that the other tests don't start running until the Collector is stood up? On an underpowered machine, this could take a while, and I wouldn't think we would want that to count against the other tests.